tgui/packages/tgui/interfaces/SDQLSpellMenu.js

```diff
@@ -0,0 +1,476 @@
+          width="75%"
+          fluid
+          value={entry[1].value}
+          onChange={(e, value) => act('list_variable_change_value', { list: list, name: entry[0], value: value })} />
+      );
+    case "list":
+      return (
+        <SDQLSpellListEntry list={list+"/"+entry[0]} />
+      );
+    default:
+      return (
+        <Box bold textColor="bad">
+          {"You shouldn't be seeing this!"}
+        </Box>
+      );
+  }
+};
+
+const SDQLSpellIcons = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+    saved_vars,
+    type,
+    action_icon,
+    hand_icon,
+    projectile_icon,
+    overlay_icon,
+    mouse_icon,
+  } = data;
+
+  const vars = typevars(type);
+
+  return (
+    <Section fill>
+      <Stack vertical>
+        {type && (
+          <Section title="Action Button Icon">
+            <Box
+              as="img"
+              height="64px"
+              width="auto"
+              m={0}
+              src={`data:image/jpeg;base64,${action_icon}`}
+              style={{
+                '-ms-interpolation-mode': 'nearest-neighbor',
+              }} />
+          </Section>
+        )}
+        {type === "targeted/touch" && (
+          <Section title="Touch Attack Icon">
+            <Box
+              as="img"
+              height="64px"
+              width="auto"
+              m={0}
+              src={`data:image/jpeg;base64,${hand_icon}`}
+              style={{
+                '-ms-interpolation-mode': 'nearest-neighbor',
+              }} />
+          </Section>
+        )}
+        {type === "aimed" && (
+          <Section title="Projectile Icon">
+            <Box
+              as="img"
+              height="64px"
+              width="auto"
+              m={0}
+              src={`data:image/jpeg;base64,${projectile_icon}`}
+              style={{
+                '-ms-interpolation-mode': 'nearest-neighbor',
+              }} />
+          </Section>
+        )}
+        {(type && vars.some(entry => entry.name === "ranged_mousepointer") && saved_vars["ranged_mousepointer"]) && (
+          <Section title="Mouse Cursor">
+            <Box
+              as="img"
+              height="64px"
+              width="auto"
+              m={0}
+              src={`data:image/jpeg;base64,${mouse_icon}`}
+              style={{
+                '-ms-interpolation-mode': 'nearest-neighbor',
+              }} />
+          </Section>
+        )}
+        {(type && (Object.prototype.hasOwnProperty.call(saved_vars, "overlay") && saved_vars["overlay"] === 1)) && (
+          <Section title="Overlay Icon">
+            <Box
+              as="img"
+              height="64px"
+              width="auto"
+              m={0}
+              src={`data:image/jpeg;base64,${overlay_icon}`}
+              style={{
+                '-ms-interpolation-mode': 'nearest-neighbor',
+              }} />
+          </Section>
+        )}
+      </Stack>
+    </Section>
+  );
+};
```

Comment:
Missing newline........ Are you still not using editorconfig plugin in vscode? How many other plugins do you also not use?

Reply:
I'm not using ESLint because I have no idea how to install all the necessary node packages in a way the plugin can recognize.

Reply:
Just install them in vscode.

https://github.com/goonstation/goonstation/blob/master/.github/DEV_GUIDE.md#step-4-extension-o-rama-gear

Reply:
Failed to load parser '@typescript-eslint/parser' declared in '.eslintrc.yml': Cannot find module '@typescript-eslint/parser' Require stack: - C:\Users\kayyc\tgstation\tgui\.eslintrc.yml

Reply:
Failed to load parser '@typescript-eslint/parser' declared in '.eslintrc.yml': Cannot find module '@typescript-eslint/parser' Require stack: - C:\Users\[CENSORED]\tgstation\tgui\.eslintrc.yml